R/profile_wrapper.R

```diff
@@ -56,21 +55,31 @@ profile_wrapper <- function(mydir, model_settings) {
     ), file = "run_diag_warning.txt")
     message(paste0("Running profile for ", para, "."))
 
-    # Copy the control file to run from the copy
-    if (!file.exists(file.path(profile_dir, "control.ss_new"))) {
-      orig_dir <- getwd()
-      setwd(profile_dir)
-      r4ss::run(
-        dir = profile_dir,
-        exe = model_settings$exe,
-        extras = model_settings$extras
-      )
-      setwd(orig_dir)
+    # check for whether oldctlfile exists
+    if (!file.exists(file.path(profile_dir, model_settings$oldctlfile))) {
+      # if the oldctlfile is control.ss_new, and doesn't exist,
+      # run the model to create it
+      if (model_settings$oldctlfile == "control.ss_new") {
+        if (model_settings$verbose) {
+          "running model to get control.ss_new file"
```

Comment:
:bug: I am pretty sure that this line should use `message()` rather than just containing a character string.

Reply:
oops, that's definitely my fault